crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2067,6 +2067,16 @@ impl<'src> Parser<'src> {
             }
         }
 
+        if with_item_kind == WithItemKind::Parenthesized && !self.at(TokenKind::Rpar) {
+            // test_err with_items_parenthesized_missing_comma
+            // with (item1 item2): ...
+            // with (item1 as f1 item2): ...
+            // with (item1, item2 item3, item4): ...
+            // with (item1, item2 as f1 item3, item4): ...
+            // with (item1, item2: ...
+            self.expect(TokenKind::Comma);
+        }
```

Comment:
This is only the case for the first "gap" as any other missing commas will be highlighted by the list parsing in `parse_with_items`.